components/iam/pkg/server/server.go

```diff
@@ -24,9 +28,30 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
 		return fmt.Errorf("failed to initialize iam server: %w", err)
 	}
 
+	err = register(srv)
+	if err != nil {
+		return fmt.Errorf("failed to register services to iam server")
+	}
+
 	if listenErr := srv.ListenAndServe(); listenErr != nil {
 		return fmt.Errorf("failed to serve iam server: %w", err)
 	}
 
 	return nil
 }
+
+func register(srv *baseserver.Server) error {
+	router := mux.NewRouter()
+	mount(router, "/oidc", oidc.Router())
+
+	srv.HTTPMux().Handle("/", router)
```

Comment:
Is there a reason to not extend `baseserver.Server` for that?

Reply:
`baseserver.Server` as such doesn't need Chi or gorilla for that matter. In go, all http handlers are ultimately an `http.Handler` and for that reason, we make our choices on the application level and still remain compatible with `http.Handler`. That also means that we shouldn't push this into baseserver as we'd be dictating the use of a "helper" library when the more general interface sufficies.